presto-testing-docker/src/main/java/io/prestosql/testing/TemporaryDockerContainer.java

```diff
@@ -0,0 +1,215 @@
+    }
+
+    private void setUp(List<Integer> ports)
+            throws Exception
+    {
+        dockerClient = DefaultDockerClient.fromEnv().build();
+        if (!image.endsWith("-SNAPSHOT") && dockerClient.listImages(DockerClient.ListImagesParam.byName(image)).isEmpty()) {
+            dockerClient.pull(image);
+        }
+        if (isDebug()) {
+            dockerClient.listContainers().stream()
+                    .filter(container -> container.image().equals(image))
+                    .forEach(container -> {
+                        checkState(containerId == null, "There are more than single container running");
+                        this.containerId = container.id();
+                    });
+            if (containerId != null) {
+                LOG.info("%s container already exists with id: %s", hostName, containerId);
+            }
+            else {
+                createContainer(ports);
+            }
+            if (hostPorts == null) {
+                calculateHostPorts(ports);
+            }
+        }
+        else {
+            dockerClient.listContainers().stream()
+                    .filter(container -> container.image().equals(image))
+                    .map(Container::id)
+                    .forEach(this::removeContainer);
+            createContainer(ports);
+        }
+
+        LOG.info("Auto-assigned host ports are %s", hostPorts);
+    }
+
+    public int getHostPort(int port)
+    {
+        checkArgument(hostPorts.keySet().contains(port), "Port %s is not bound", port);
+        return hostPorts.get(port);
+    }
+
+    private void createContainer(List<Integer> ports)
+            throws Exception
+    {
+        Map<String, List<PortBinding>> portBindings = ports.stream()
+                .collect(toImmutableMap(Object::toString, port -> ImmutableList.of(PortBinding.create(hostIp, "0"))));
+        String[] exposedPorts = ports.stream()
+                .map(Object::toString)
+                .collect(toImmutableList())
+                .toArray(new String[ports.size()]);
+
+        containerId = dockerClient.createContainer(ContainerConfig.builder()
+                .hostConfig(HostConfig.builder()
+                        .portBindings(portBindings)
+                        .build())
+                .exposedPorts(exposedPorts)
+                .env(environment.entrySet().stream()
+                        .map(entry -> format("%s=%s", entry.getKey(), entry.getValue()))
+                        .collect(toImmutableList()))
+                .hostname(hostName)
+                .image(image)
+                .build())
+                .id();
+
+        LOG.info("Started docker container with id: %s", containerId);
+
+        dockerClient.startContainer(containerId);
+
+        calculateHostPorts(ports);
+
+        ports.forEach(port -> {
+            RetryPolicy<Object> retryPolicy = new RetryPolicy<>()
+                    .withMaxDuration(Duration.of(5, MINUTES))
```

Comment:
This is the second place where startup timeout is defined. Please make it into a constant.